tests/tick.test.mjs

```diff
@@ -659,4 +659,102 @@ describe('Tick', async () => {
       },
     });
   });
+
+  it('should reset to baseRegistrationFee when demandFactor is 0.5 for consecutive epochs', async () => {
+    const genesisEpochStart = 1722837600000 + 1;
+    const epochDurationMs = 60 * 1000 * 60 * 24; // 24 hours
+    const distributionDelayMs = 60 * 1000 * 40; // 40 minutes (~ 20 arweave blocks)
+
+    const zeroEpochTick = await handle(
+      {
+        Tags: [{ name: 'Action', value: 'Tick' }],
+        Timestamp: genesisEpochStart,
+      },
+      startMemory,
+    );
+
+    const getRegistrationFeesResult = await handle(
+      {
+        Tags: [{ name: 'Action', value: 'Get-Registration-Fees' }],
+      },
+      zeroEpochTick.Memory,
+    );
+    const baseFeeAtZeroEpoch = JSON.parse(
+      getRegistrationFeesResult.Messages[0].Data,
+    )['9']['lease']['1'];
+    assert.equal(baseFeeAtZeroEpoch, 600_000_000);
+
+    const fundedUser = 'funded-user-'.padEnd(43, '1');
+    const transferMemory = await transfer({
+      recipient: fundedUser,
+      quantity: 100_000_000_000_000,
+      memory: zeroEpochTick.Memory,
+    });
+
+    // Buy records in this epoch
+    let buyRecordMemory = transferMemory;
+    for (let i = 0; i < 10; i++) {
+      const buyRecordResult = await handle(
+        {
+          Tags: [
+            { name: 'Action', value: 'Buy-Record' },
+            { name: 'Name', value: 'test-name-' + i },
+            { name: 'Purchase-Type', value: 'permabuy' },
+          ],
+        },
+        buyRecordMemory,
+      );
+      buyRecordMemory = buyRecordResult.Memory;
+    }
```

Comment:
nit: you don't have to buy records to validate the base fees drop after consecutive periods at the minimum

I recommend splitting this up into two tweets
1. checks fees during an epoch, buy a few records, and check base fees again (they should not change) and validate the demand factor has increased correctly
2. check base fees at first epoch, tick the consecutive number of epochs required to reduce base registration fees without any name purchases, validate the base fees dropped correctly compared to the original

Reply:
86060a2